ChurchCRM/CRM
# Standardize Birthdate Formatting

#### What's this PR do?
We've gone through many iterations, and birthdates seem to have always been a problem.  This PR standardizes, wherever possible, the formatting of birthrates into the Person ORM object.

It also moves birthday related functions out of the deprecated ```Functions.php``` file into the namespaced MiscUtils file for easier future refactoring of code that is not currently using the ORM model.


#### Screenshots (if appropriate)

#### What Issues does it Close?

Closes #3804 
Closes #2340

#### What are the relevant tickets?
#3168 
#3052 
#1192
#919
#72
#41

#### Any background context you want to provide?

#### Where should the reviewer start?

#### How should this be manually tested?

*Person View*
1) Open a person profile. Populate the birthday
2) Notice the full m/d/Y birthday is printed if all data is available
3) Remove the Year.  Notice that only month and day are printed
4) re-add the year.  Click "Hide Age" flag.  Notice only month and day are printed
5) check the "Family" tab of the person view and ensure that all family members birth dates appear correctly according to "Hide age" or missing birth year.

*Family View* 
1) On a Family View page with multiple members, ensure that all family members birth dates appear correctly - honor "hide age" and/or missing birth year

*Directory*
1) Repeat the above procedure for Person View, but generate a new PDF directory after changing the dates.  Ensure appropriate person dates are in the PDF
2) Repeat the previous step for both persons who are members of a multiperson family and a single person family.  Note that single person families have the person's birthdate in the record header instead of the body.

*Sunday School Dashboard*
1) Repeat the Person View process on a ```FamilyRole=child``` who has not been assigned to a Sundayschool class, and ensure that their birth date & age is represented correctly on the Sunday school dashboard

*Sunday School Class View*
1) Repeat the Person View process on a ```FamilyRole=child``` who has been assigned to a Sunday School class, and ensure that their birth date & age is represented correctly on the Sunday class view

*Export CSV*
1)  Export a CSV form the cart (or anywhere else).  Be sure to "check" the "Age / Years Married" and "Birth / Anniversary Date" checkboxes on the export fields list.
2) ensure that birth dates are month and day for missing year / hide age people 
3) ensure that birth dates and ages are correct for people who have a birth year and who do not have "hide age" checked.

#### How should the automated tests treat this?

#### Questions:
- Is there a related website / article to substantiate / explain this change?

- Does the development wiki need an update?

- Does the user documentation wiki need an update?

- Does this add new dependencies?

- Does this need to add new data to the demo database
